### PR TITLE
feat(imagebrush): add support for RelativeTransform on skia

### LINKED ISSUE #16775: Support `RelativeTransform` for `ImageBrush` on Skia
### What would you like to be added

Currently, this property is ignored on Skia (for `ImageBrush` at least).

See samples `ImageBrushWithRotateTransform` and `BorderImageBrushRelativeTransform`

### Why is this needed

_No response_

### For which platform

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by ramezgerges**
GitHub Issue (If applicable): closes #16775

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI.Composition/Composition/CompositionSurfaceBrush.skia.cs [hunk 1]
```diff
@@ -88,31 +90,48 @@ internal override void UpdatePaint(SKPaint fillPaint, SKRect bounds)
...
+					matrix *= Matrix3x2.CreateScale(bounds.Width, bounds.Height);
+
+					// The image rescaling (i.e resampling) algorithm in the shader directly is really low quality (but really fast).
+					// SkiaSharp 3 introduces new SKSamplingOptions, but for now there is no alternative to writing
+					// the shader logic by hand.
+					// When we move to SkiaSharp3, remove this
+					// ******************************************************************************
+					if (_effect is null)
+					{
+						_effect = SKRuntimeEffect.Create(ImageResamplingShader, out var error);
```
Comment (Youssef1313) [inline src/Uno.UI.Composition/Composition/CompositionSurfaceBrush.skia.cs:115]: Could this be cached to a static field?

Comment (Youssef1313) [reply src/Uno.UI.Composition/Composition/CompositionSurfaceBrush.skia.cs:115]: Oh you already do.

File: src/Uno.UI.Composition/Composition/CompositionSurfaceBrush.skia.cs [hunk 2]
```diff
@@ -106,31 +106,9 @@ internal override void UpdatePaint(SKPaint fillPaint, SKRect bounds)
...
-					{
-						["image"] = SKShader.CreateImage(scs.Image, SKShaderTileMode.Decal, SKShaderTileMode.Decal)
-					};
-
-					var imageShader = _effect.ToShader(false, uniforms, children, matrix.ToSKMatrix());
-					// ******************************************************************************
-					// and replace with this one line (and maybe add mipmapping
+					// There is no sound workaround for this at the moment. See https://github.com/unoplatform/uno/issues/17325.
+					var imageShader = SKShader.CreateImage(scs.Image, SKShaderTileMode.Decal, SKShaderTileMode.Decal, matrix.ToSKMatrix());
+					// SkiaSharp 3 introduces new SKSamplingOptions. When we move to SkiaSharp 3, replace the line about with this one.
```
Comment (Youssef1313) [inline src/Uno.UI.Composition/Composition/CompositionSurfaceBrush.skia.cs:111]: With SkiaSharp 3, we'll be able to have the good quality and also keeping everything working as intended?

Comment (ramezgerges) [reply src/Uno.UI.Composition/Composition/CompositionSurfaceBrush.skia.cs:111]: Yes.

**Review by jeromelaban — COMMENTED** [review]
@ramezgerges Nice! Could you add some visual tests for this?

**Comment by ramezgerges** [comment]
@jeromelaban There are already the 2 samples mentioned in the issue (ImageBrushWithRotateTransform and BorderImageBrushRelativeTransform). Any particular additional sample you have in mind?

**Comment by jeromelaban** [comment]
@ramezgerges not particularly, but as you're adding a new behavior, this would imply that there's specific regression testing that needs to be included.

**Review by jeromelaban — APPROVED** [review]

**Pull request merged**